src/opt.c

```diff
@@ -103,6 +104,71 @@ bool opt_get_verbose(opt_name_t name, const char *filter)
       return strstr(filter, value) != NULL;
 }
 
+int opt_parse_comma_separated(const char *opt, const char *optarg,
+                              opt_separed_t *allowed, int allowed_cnt,
+                              char **wc_buf)
+{
+   assert(allowed_cnt > 1);
+
+   char buf[64] = {0};
```

Comment:
You don't need to copy the option to a separate buffer. Just keep a pointer to the start of the option name and then use `strncmp`. Something like:

```c
   int rv = 0;
   const char *start = optarg;

   while (1) {

      // Process buffer gathered so far, search if any allowed options match
      if (*optarg == ',' || *optarg == '\0') {
         opt_separed_t *curr_opt;
         for (int i = 0; i < allowed_cnt; i++) {
            curr_opt = &(allowed[i]);
            const char *wc = strchr(curr_opt->opt, '*');
            int wc_pos =  wc - curr_opt->opt;

            // Check for wild-card options -> Copy matched end to wildcard buffer
            if (wc && !strncmp(curr_opt->opt, start, wc_pos)) {
               strcpy(*wc_buf, start + wc_pos);
               wc_buf++;
               break;
            }
            // Check for regular options
            else if (!strncmp(curr_opt->opt, start, optarg - start)) {
               break;
            }
            if (i == allowed_cnt - 1)
               curr_opt = NULL;
         }

         if (curr_opt == NULL) {
            /*  .... */
         } else
            rv |= curr_opt->mask;

         if (*optarg == '\0')
            break;

         start = optarg + 1;
      }
      optarg++;
   }
```